js-miniapp-sdk/src/miniapp-error-type.ts

```diff
@@ -0,0 +1,99 @@
+interface MiniAppJson {
+  message: string;
+  type: string;
+}
+
+export function parseMiniAppError(jsonString: string): MiniAppJson {
+  return JSON.parse(jsonString);
+}
+
+/**
+ * This class is a representation of an error sent from MiniApp mobile SDK
+ */
+export class MiniAppError extends Error {
+  /**
+   * the raw input received from the SDK used to construct the MiniAppError
+   */
+  raw: string;
+  customMessage: string;
+
+  constructor(public errorInput: MiniAppJson) {
+    super();
+    this.raw = JSON.stringify(errorInput);
+    this.name = errorInput.type;
+    this.setMessage(errorInput.message);
+  }
+
+  setMessage(newMessage: string) {
+    this.message = newMessage;
+    this.customMessage = newMessage;
+  }
+
+  /**
+   * Takes the error input string sent from mobile SDK.
+   * Should be formatted as follow: "error_key: error message"
+   *
+   * @param errorInput error input string sent from mobile SDK
+   */
+  static fromCustomString(errorInput: string): MiniAppError {
```

Comment:
See my message below about moving the parsing to the bridge module. In that case, I don't think there's a need to handle the case where the error is formatted as `key: message` since the parsing is bundled with the Android/iOS SDK.

Reply:
I don't even use this method anymore, I will just remove it now we are in the bridge